agent/discover/discover.go

```diff
@@ -0,0 +1,31 @@
+package discover
+
+import (
+	"fmt"
+	"log"
+
+	consul "github.com/hashicorp/consul/api"
+)
+
+type Discovery func() (Discoverer, error)
+
+type Discoverer interface {
+	// this function checks whether this particular implementation of Discover
+	// is relevant for this node. It is used as a gating condition for other
+	// functionality in this implementation
+	shouldDiscover(client consul.Client) bool
+	// Provide a Consul Key-Value Store Prefix that is used to store individual
+	// Key-Values for this Cluster
+	KVStorePrefix() string
+}
+
+// Create or Updating the given Consul Key-Value Path Store with a new value from the Agent
+func StoreDiscovery(discover Discoverer, client consul.Client, cStorePath, cStoreValue string) {
```

Comment:
This looks a pretty generic function that will be used outside of discoveries most probably

Reply:
no, it uses the Discoverer Interface implementations, so it isn't useable outside discoveries.